site/pages/api/pledge/index.ts

```diff
@@ -8,16 +9,17 @@ export default async function handler(
   switch (req.method) {
     case "PUT":
       try {
-        const sessionToken = req.headers.authorization?.split(" ")[1];
-        const data = await findOrCreatePledge({
-          pledge: req.body,
-          sessionToken,
+        const signature = req.headers.authorization?.split(" ")[1] as string;
+        const pledge = await findOrCreatePledge({
+          pageAddress: req.body.pageAddress,
+          pledge: req.body.pledge,
+          signature,
         });
-        const pledge = JSON.parse(JSON.stringify(data));
 
         res.status(200).json({ pledge: pledgeResolver(pledge) });
       } catch (error) {
-        console.log(error);
+        console.error(error);
+        res.status(400).json({ error });
```

Comment:
@Atmosfearful
How would we type server errors and expose the errors to the user. Or would we prefer a generic server error message?

Reply:
just error.message json, return `500` for generic `Internal Server Error`
```
  } catch ({ message }) {
    console.error("Request failed:", message);
    res.status(500).json({ message });
  }
  ```

Reply:
This let's us write 
`throw new Error("Helpful error message for server and client debugging!!!!");`
anywhere within the handler